check_haproxy.rb

```diff
@@ -4,12 +4,18 @@
 require 'open-uri'
 require 'ostruct'
 require 'csv'
+require 'openssl'
 
 OK = 0
 WARNING = 1
 CRITICAL = 2
 UNKNOWN = 3
 
+# allows https with invalid certificate on ruby 1.8+
+#
+# src: also://snippets.aktagon.com/snippets/370-hack-for-using-openuri-with-ssl
+OpenSSL::SSL::VERIFY_PEER = OpenSSL::SSL::VERIFY_NONE
```

Comment:
this should only be set as part of a command-line option and should not be the default